config/webpack/plugin/TerraDevSite.js

```diff
@@ -0,0 +1,57 @@
+const path = require('path');
+const { DefinePlugin } = require('webpack');
+const SetupPlugin = require('./SetupPlugin');
+const GeneratePlugin = require('./GeneratePlugin');
+const loadSiteConfig = require('../../../scripts/generate-app-config/loadSiteConfig');
+
+/**
+ * Generate a terra-dev-site
+ */
+class TerraDevSite {
+  constructor({ env = {}, sites = [] } = {}) {
+    // Sites to generate. Add the default site then load the config if not present
+    this.sites = [
+      {
+        configFileName: 'site.config.js',
+        defaultConfigPath: path.resolve(__dirname, '..', '..', 'site', 'site.config.js'),
+        indexPath: path.resolve(__dirname, '..', '..', '..', 'lib', 'TerraDevSite'),
+      },
+      ...sites,
+    ].map((site) => {
+      const siteConfig = site.siteConfig || loadSiteConfig(site.configFileName, site.defaultConfigPath);
+      const locale = env.defaultLocale;
+      if (locale) {
+        siteConfig.appConfig.defaultLocale = locale;
+      }
+      return ({
+        ...site,
+        // load config if siteConfig is not already defined.
+        siteConfig,
+      });
+    });
+  }
+
+  apply(compiler) {
+    // Use default public path else the env else /
```

Comment:
Is this actually if default, else '/', else env?

Reply:
Default || env || '/'
I'll rework the code to make it more clear.